### PR TITLE
Fixed #31195 -- Relaxed GeometryDistance test for PROJ 5.2+.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/31195

### CODE REVIEW

**Comment by felixxm** [comment]
@sir-sigurd I couldn't find [PROJ 5.3](https://proj.org/download.html#download). Did you mean 5.2+?

File: tests/gis_tests/geoapp/test_functions.py [hunk 1]
```diff
@@ -281,7 +281,9 @@ def test_geometry_distance(self):
             14.713098433352,
             34.3635252198568,
             276.987855073372,
-        ])
+        )
+        for city, expected_distance in zip(qs, distances):
+            self.assertAlmostEqual(city.distance, expected_distance)
```
Comment (pope1ni) [inline tests/gis_tests/geoapp/test_functions.py]: `with self.subTest(…):`

**Comment by sir-sigurd** [comment]
> @sir-sigurd I couldn't find [PROJ 5.3](https://proj.org/download.html#download). Did you mean 5.2+?

Yes, thanks.

File: tests/gis_tests/geoapp/test_functions.py [hunk 2]
```diff
@@ -281,7 +281,10 @@ def test_geometry_distance(self):
             14.713098433352,
             34.3635252198568,
             276.987855073372,
-        ])
+        )
+        for city, expected_distance in zip(qs, distances):
+            with self.subTest(distance=city.distance):
```
Comment (sir-sigurd) [inline tests/gis_tests/geoapp/test_functions.py]:
```suggestion
            with self.subTest(city=city):
```

**Pull request merged**